peer/mrunoncemap_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2015 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package peer
+
+import (
+	"fmt"
+	"testing"
+)
+
+// TestMruNonceMap ensures the mruNonceMap behaves as expected including
+// limiting, eviction of least-recently used entries, specific entry removal,
+// and existence tests.
+func TestMruNonceMap(t *testing.T) {
+	// Create a bunch of fake nonces to use in testing the mru nonce code.
+	numNonces := 10
+	nonces := make([]uint64, 0, numNonces)
+	for i := 0; i < numNonces; i++ {
+		nonces = append(nonces, uint64(i))
+	}
+
+	tests := []struct {
+		name  string
+		limit int
+	}{
+		{name: "limit 0", limit: 0},
+		{name: "limit 1", limit: 1},
+		{name: "limit 5", limit: 5},
+		{name: "limit 7", limit: 7},
+		{name: "limit one less than available", limit: numNonces - 1},
+		{name: "limit all available", limit: numNonces},
+	}
+
+testLoop:
+	for i, test := range tests {
+		// Create a new mru nonce map limited by the specified test
+		// limit and add all of the test nonces.  This will cause
+		// evicition since there are more test nonces than the limits.
+		mruNonceMap := newMruNonceMap(uint(test.limit))
+		for j := 0; j < numNonces; j++ {
+			mruNonceMap.Add(nonces[j])
+		}
+
+		// Ensure the limited number of most recent entries in the list
+		// exist.
+		for j := numNonces - 1; j >= numNonces-test.limit; j-- {
```

Comment:
Can this and the loop below start iterate with increasing j's?  For this one it should start at the limit and iterate up to numNonces, and the one below would iterate from 0 up to the limit. imo that would be more readable than iterating them backwards.